File: Easy/408_Valid_word_Abbreviation.cpp
class Solution {
public:
    bool validWordAbbreviation(string word, string abbr) {
        int pos=0;
        int number;
        string numberstring;
        int i;
        for(i=0;i<abbr.size();i++)
        {
            if(abbr[i]<='z'&&abbr[i]>='a')
            {
                if(word[pos]!=abbr[i])
                return false;
                pos++;
                 //printf("%d",i);
            }
            if(abbr[i]<='9'&&abbr[i]>='0')
            {
                if(abbr[i]=='0')
                return false;
                numberstring.push_back(abbr[i]);
                
                int j=i+1;
                while(j<abbr.size()&&(abbr[j]<='9'&&abbr[j]>='0'))
                {
                    numberstring.push_back(abbr[j]);
                    j++;
                }
                i=j-1;
                pos=pos+(stoi(numberstring));
              // printf("%d",pos);
                numberstring.clear();
                if(pos>word.size())
                return false;
            }
        }
       // printf("%d %d",i,pos);
        if(word.size()!=pos)
        return false;
        return true;
    }
};

File: Easy_python/408_valid_word_abbreviation.py
class Solution:
    def validWordAbbreviation(self, word: str, abbr: str) -> bool:
        countlist=[]
        i=0
        while i< len(abbr):
            if abbr[i]<='z' and abbr[i]>='a':
                countlist.append(abbr[i])
                i=i+1
            else:
                temp=""
                while i<len(abbr) and (abbr[i]>='0' and abbr[i]<='9'):
                    temp=temp+abbr[i]
                    i=i+1
                    if temp[0]=='0':
                        return False
                countlist.append(temp)
        currentpos=0
        flag=0
        for i in range(len(countlist)):
            if currentpos>=len(word):
                flag=1
                break
            if len(countlist[i])>1 or (countlist[i]>"z" or countlist[i]<"a"):
                currentpos=currentpos+int(countlist[i])
            else:
                if word[currentpos]!=countlist[i]:
                    return False
                else:
                    currentpos=currentpos+1
            
        if currentpos!=len(word) or flag==1:
            return False   
        else:
            return True
